src/pages/workspaces/workspace/tools/LaunchAnalysisModal.js

```diff
@@ -1,178 +1,116 @@
 import _ from 'lodash/fp'
-import { Fragment } from 'react'
 import { div, h } from 'react-hyperscript-helpers'
-import { AutoSizer } from 'react-virtualized'
-import { buttonPrimary, link, search } from 'src/components/common'
-import { centeredSpinner } from 'src/components/icons'
+import { buttonPrimary } from 'src/components/common'
+import { spinner } from 'src/components/icons'
 import Modal from 'src/components/Modal'
-import TabBar from 'src/components/TabBar'
-import { GridTable, HeaderCell, TextCell } from 'src/components/table'
 import { ajaxCaller } from 'src/libs/ajax'
 import colors from 'src/libs/colors'
-import { renderDataCell } from 'src/libs/data-utils'
-import * as Utils from 'src/libs/utils'
 import { Component } from 'src/libs/wrapped-components'
+import EntitySelectionType from 'src/pages/workspaces/workspace/tools/EntitySelectionType'
 
 
 export default ajaxCaller(class LaunchAnalysisModal extends Component {
-  constructor(props) {
-    super(props)
-
-    this.state = { filterText: '', entityType: props.config.rootEntityType }
-  }
-
   render() {
     const { onDismiss } = this.props
-    const { entityType, entityMetadata, entities, attributeFailure, entityFailure, filterText, launching, selectedEntity } = this.state
-    const { attributeNames } = entityMetadata ? entityMetadata[entityType] : {}
+    const { launching, message, launchError } = this.state
 
-    return h(Modal, _.isUndefined(entityType) ? {
-      title: 'Launching Analysis', showCancel: false, okButton: false
-    } : {
+    return h(Modal, {
```

Comment:
Yes, I'd like there to be more information in there, such as how many jobs will actually be launched. The code needed to determine that number in all cases will require some further refactoring. The necessary changes weren't obvious to me at the time but maybe I'll be able to see it now that I've stepped away from it for a bit. If not, I'll file an issue for it.

Reply:
I think it's probably better to just file a ticket for it since this is already a pretty hefty PR. Good to note that we should continue to improve on it though.

Reply:
Created a ticket for it in #1203.